### PR TITLE
`rustc_codegen_llvm` cleanups

### PR DESCRIPTION
**Pull request by nnethercote**
Some improvements I found while reading through this crate's code.

r? @michaelwoerister

### CODE REVIEW

File: compiler/rustc_codegen_llvm/src/builder.rs [hunk 1]
```diff
@@ -305,84 +317,17 @@ impl<'a, 'll, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
...
-    }
-
-    fn frem_algebraic(&mut self, lhs: &'ll Value, rhs: &'ll Value) -> &'ll Value {
-        unsafe {
-            let instr = llvm::LLVMBuildFRem(self.llbuilder, lhs, rhs, UNNAMED);
-            llvm::LLVMRustSetAlgebraicMath(instr);
-            instr
-        }
+    set_math_builder_methods! {
+        fadd_fast(x, y) => (LLVMBuildFAdd, LLVMRustSetFastMath),
```
Comment (jieyouxu) [inline compiler/rustc_codegen_llvm/src/builder.rs:321]: Remark: I do like how these look :+1:

**Review by jieyouxu — APPROVED** [review]
AFAICT the changes look good, and they do make the code nicer to read 👍 Some of the changes are a bit more subtle (like the hidden visibility logic), so I would like someone else to also take a look.

**Comment by nnethercote** [comment]
The hidden visibility commit is definitely the trickiest one! I have fixed the conflicts and updated.

**Review by jieyouxu — APPROVED** [review]
Thanks! I looked at the hidden visibility changes again, and I think the restructure is correct (as in I think it does preserve the original logic). Please r=me after PR CI is green.

**Pull request merged**